specs-derive/src/impl_saveload.rs

```diff
@@ -175,30 +172,52 @@ fn saveload_named_struct(
     name: &Ident,
     saveload_name: &Ident,
     generics: &Generics,
-    saveload_fields: &[Field],
+    saveload_fields: &[FieldMetaData],
 ) -> (TokenStream, TokenStream, TokenStream) {
     let (_, ty_generics, where_clause) = generics.split_for_impl();
 
+    let fields = saveload_fields.iter().map(|f| f.field.clone());
```

Comment:
does `fields` need to be owned? If I remember correctly, they should be able to be converted into tokens even if they are `&Field`s.

oh, if it's to do with lifetimes, wrapping it in `{}` should work:

```rust
let struct_def = {
    let fields = saveload_fields.iter().map(|f| f.field);

    quote! {
        struct #saveload_name #ty_generics #where_clause {
            #( #fields ),*
        }
    };
};
```